tests/testthat/test-repo.R

```diff
@@ -91,3 +91,33 @@ test_that("returns NULL if no urls found", {
   desc <- desc::desc(text = "URL: https://pkgdown.r-lib.org")
   expect_equal(package_repo(desc, list()), NULL)
 })
+
+test_that("repo_type detects repo type", {
+  expect_equal(
+    repo_type(
+      list(repo = list(url = list(home = "https://github.com/r-lib/pkgdown")))
```

Comment:
I'd make a little helper for this:

```R
repo_type2 <- function(url) {
   repo_type(list(repo = list(url = list(home = url)))
}
```

Then all the individual tests can fit on one line